pennylane_lightning/src/Gates.cpp

```diff
@@ -556,6 +556,52 @@ void Pennylane::CSWAPGate::applyKernel(const StateVector& state, const std::vect
 
 // -------------------------------------------------------------------------------------------------------------
 
+const string Pennylane::Unitary::label = "QubitUnitary";
+
+Pennylane::Unitary Pennylane::Unitary::create(const vector<double>& parameters) {
+    auto size = parameters.size();
+    unsigned int log2size = 0;
+    while (size >>= 1) ++log2size;
```

Comment:
Personally, would not mind having this still be a `while` with syntax of
```cpp
while (size >>= 1){
    ++log2size;
}
```
Also, this functionality could potentially come in handy in other parts of the code base too, could consider defining it in `Utils.hpp`.